AK/SIMDExtras.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2021, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/SIMD.h>
+
+namespace AK::SIMD {
+
+// SIMD Vector Expansion
+
+ALWAYS_INLINE static constexpr f32x4 expand4(float f)
+{
+    return f32x4 { f, f, f, f };
+}
+
+ALWAYS_INLINE static constexpr i32x4 expand4(i32 i)
+{
+    return i32x4 { i, i, i, i };
+}
+
+ALWAYS_INLINE static constexpr u32x4 expand4(u32 u)
+{
+    return u32x4 { u, u, u, u };
+}
+
+// Casting
+
+template<typename TSrc>
+ALWAYS_INLINE static u32x4 to_u32x4(TSrc v)
+{
+    return __builtin_convertvector(v, u32x4);
+}
+
+template<typename TSrc>
+ALWAYS_INLINE static i32x4 to_i32x4(TSrc v)
+{
+    return __builtin_convertvector(v, i32x4);
+}
+
+template<typename TSrc>
+ALWAYS_INLINE static f32x4 to_f32x4(TSrc v)
+{
+    return __builtin_convertvector(v, f32x4);
+}
+
+// Rounding
+
+ALWAYS_INLINE static f32x4 truncate(f32x4 v)
+{
+    return to_f32x4(to_i32x4(v));
+}
+
+ALWAYS_INLINE static f32x4 floor(f32x4 v)
+{
+    v = (v < 0.f) ? v - 1 : v;
+    return truncate(v);
+}
+
+ALWAYS_INLINE static f32x4 ceil(f32x4 v)
+{
+    v = (v >= 0.f) ? v + 1 : v;
+    return truncate(v);
+}
```

Comment:
I guess that's one way to do it,
might get problems with values above/below +-INT_MAX,
but within libGL this should suffice

Reply:
That's a valid point. It might be better then to name these something like `fast_floor` or `approximate_floor` if we expose these functions to the whole system :thinking: Also `ceil` should check for `v > 0.f` as `ceil(0)` is 0 and not 1.

Reply:
Renamed the functions to `floor_int_range()` etc.